VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionComment.java

```diff
@@ -142,7 +142,17 @@ public String getAuthorLabel() {
                 public Date getCreatedAt() {
                     return endorsedAt;
                 }
+
+                @Override
+                public boolean isAuthorAnonymous() {
+                    return DiscussionComment.this.isAuthorAnonymous();
```

Comment:
This should check the `endorsedBy` field I guess, since that is what is displayed in endorsed responses.